cmd/insprd/memory/tree/dapp_utils_test.go

```diff
@@ -1242,3 +1243,40 @@ func Test_validAliases(t *testing.T) {
 		})
 	}
 }
+
+func TestSelectBrokerFromPriorityList(t *testing.T) {
+	defer brokers.ResetBrokerMemory()
+	type args struct {
+		brokerList []string
+	}
+	tests := []struct {
+		name   string
+		args   args
+		want   string
+		before func()
+	}{
+		{
+			name: "Should return the first available broker",
+			args: args{
+				brokerList: []string{"A", "Broker_B"},
+			},
+			want: "Broker_B",
+			before: func() {
+				bmm := brokers.GetBrokerMemory()
+				bmm.Create("Broker_A", nil)
+				bmm.Create("Broker_B", nil)
+				bmm.SetDefault("Broker_A")
+			},
+		},
```

Comment:
Nice test but there could be more tests cases to confirm that it returns the desired result.

It is a silly thing to ask but it would be nice, a few examples would be:
- it doesn't find the broker and has to return the default broker
- When passing no broker it returns the default broker
- if the log is added would be nice to confirm that is being written only when a default broker is used